lasagne/layers/autoenc.py

```diff
@@ -0,0 +1,262 @@
+                    **kwargs
+                )
+            elif isinstance(layer, DropoutLayer):
+                a_layer = DropoutLayer(
+                    incoming=incoming,
+                    p=layer.p,
+                    rescale=layer.rescale
+                )
+            elif isinstance(layer, GaussianNoiseLayer):
+                a_layer = GaussianNoiseLayer(
+                    incoming=incoming,
+                    sigma=layer.sigma
+                )
+            elif isinstance(layer, BiasLayer):
+                if b is None:
+                    kwargs = dict(b=None)
+                elif b == 'new':
+                    kwargs = dict(b=init.Constant(0.))
+                elif isinstance(b, (tuple, list)):
+                    kwargs = dict(b=b[b_idx])
+                    b_idx += 1
+                else:
+                    kwargs = dict(b=b)
+
+                a_layer = BiasLayer(
+                    incoming=incoming,
+                    **kwargs
+                )
+            else:
+                a_layer = InverseLayer(
+                    incoming=incoming,
+                    layer=layer
+                )
+            autoencoder_layers.append(a_layer)
+
+    return autoencoder_layers[-1]
+
+
+def unfold_bias_and_nonlinearity_layers(layer):
+    """
+    Given a :class:`Layer` instance representing a stacked network,
+    this function adds a :class:`NonlinearityLayer` instance in between
+    each layer with attribute nonlinearity, with the same nonlinearity,
+    while setting the nonlinearity of the original layer to the identity
+    function.
+
+    This function is useful to build autoencoders with tied parameters.
+
+    Parameters
+    ----------
+        layer: :class:`Layer` instance
+            The :class:`Layer` instance with respect to wich the new
+            stacked Neural Network with added :class:`BiasLayer`: and
+            class:`NonlinearityLayer` are built.
+
+    Returns
+    -------
+        layer: :class:`Layer` instance
+           The output :class:`Layer` of the symmetric autoencoder with
+           tied weights.
+
+    Examples
+    --------
+    >>> import lasagne
+    >>> from lasagne.layers import InputLayer, DenseLayer
+    >>> from lasagne.layers import BiasLayer, NonlinearityLayer
+    >>> from lasagne.layers import unfold_bias_and_nonlinearity_layers
+    >>> from lasagne.layers import get_all_layers
+    >>> from lasagne.nonlinearities import tanh, sigmoid, identity
+    >>> l_in = InputLayer((100, 20))
+    >>> l1 = DenseLayer(l_in, num_units=50, nonlinearity=tanh)
+    >>> l_out = DenseLayer(l1, num_units=10, nonlinearity=sigmoid)
+    >>> l_out = unfold_bias_and_nonlinearity_layers(l_out)
+    >>> all_layers = get_all_layers(l_out)
+    >>> isinstance(all_layers[0], InputLayer)
+    True
+    >>> isinstance(all_layers[1], DenseLayer)
+    True
+    >>> all_layers[1].nonlinearity == identity
+    True
+    >>> all_layers[1].b is None
+    True
```

Comment:
That style is too long for an example, and difficult to read, too. What about something like: 
```
>>> [l.__class__.__name__ for l in get_all_layers(l_out)]
['InputLayer', 'DenseLayer', 'BiasLayer', NonlinearityLayer', ...]
>>> all(l.b is None if not isinstance(l, BiasLayer))
True
```
Just some style that does not check each and every layer's attributes separately.